src/Core/Checkout/Promotion/Subscriber/PromotionIndividualCodeRedeemer.php

```diff
@@ -128,7 +128,7 @@ private function collectLineItems(EntityWrittenEvent $event): OrderLineItemColle
         $orderLineItems = new OrderLineItemCollection();
 
         foreach ($event->getWriteResults() as $result) {
-            if ($result->getPayload()['type'] !== PromotionProcessor::LINE_ITEM_TYPE) {
+            if ($result->getPayload()['type'] ?? '' !== PromotionProcessor::LINE_ITEM_TYPE) {
```

Comment:
```suggestion
            if (($result->getPayload()['type'] ?? '') !== PromotionProcessor::LINE_ITEM_TYPE) {
```

I would add braces, because I am not sure, what is interpreted first. just to be sure